packages/api-client-core/src/GadgetFunctions.ts

```diff
@@ -100,12 +100,15 @@ export interface BulkActionWithInputs<OptionsT, VariablesT> {
   <Options extends OptionsT>(inputs: VariablesT, options?: LimitToKnownKeys<Options, OptionsT>): AsyncRecord<any>;
 }
 
+export type UnionResultType = Record<string, { hasReturnType: boolean }>;
+
 export interface ActionFunctionMetadata<OptionsT, VariablesT, SelectionT, SchemaT, DefaultsT, IsBulk> {
   type: "action";
   operationName: string;
   operationReturnType?: string;
   namespace: string | string[] | null;
   modelApiIdentifier: string;
+  operatesWithRecordIdentity?: boolean;
```

Comment:
I think to be on the safe side we should make this optional and so that we don't need to add this property to pre version 1.1 apps; but let me know if you think we should make it required

Reply:
I think best to let it be optional since we can't guarantee the client has been regenerated it to offer it!